### PR TITLE
Refactor `from-token` command to use code from `autonomy.chain` module

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

This PR 

- Refactors the `from-token` command to use code from `autonomy.chain` module
- Removes unnecessary code and data files
- Updates the tests for the same

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

### CODE REVIEW

File: autonomy/chain/config.py [hunk 1]
```diff
@@ -32,12 +31,6 @@
 DEFAULT_LOCAL_CHAIN_ID = 31337
 
 
-def get_abi(filename: str) -> Dict:
-    """Service contract ABI."""
-    with (chain_constants.ABI_DIR / filename).open(mode="r", encoding="utf-8") as fp:
-        return json.load(fp=fp)
```
Comment (angrybayblade) [inline autonomy/chain/config.py:38]: No required anymore

File: autonomy/chain/constants.py [hunk 2]
```diff
@@ -29,18 +29,11 @@
...
 CONTRACTS_DIR_FRAMEWORK = DATA_DIR / CONTRACTS
 CONTRACTS_DIR_LOCAL = (
     AUTONOMY_DIR.parent / PACKAGES / VALORY / CONTRACTS
 )  # use from an editable/local installation
 
-COMPONENT_REGISTRY_ABI_FILENAME = "ComponentRegistry.json"
-AGENT_REGISTRY_ABI_FILENAME = "AgentRegistry.json"
-REGISTRIES_MANAGER_ABI_FILENAME = "RegistriesManager.json"
-SERVICE_MANAGER_ABI_FILENAME = "ServiceManager.json"
-SERVICE_REGISTRY_ABI_FILENAME = "ServiceRegistry.json"
```
Comment (angrybayblade) [inline autonomy/chain/constants.py:42]: Same

File: autonomy/cli/utils/click_utils.py [hunk 3]
```diff
@@ -73,25 +72,6 @@ def wrapper(f: Callable) -> Callable:
 
 
 def chain_selection_flag(
-    default: str = "staging", mark_default: bool = True
-) -> Callable:
-    """Flags for abci spec outputs formats."""
```
Comment (angrybayblade) [inline autonomy/cli/utils/click_utils.py:78]: Replace the old chain selection logic with the new one

File: autonomy/cli/helpers/chain.py [hunk 4]
```diff
@@ -60,7 +61,7 @@
 
 def get_ledger_and_crypto_objects(
     chain_type: ChainType,
-    keys: Path,
+    keys: Optional[Path] = None,
```
Comment (angrybayblade) [inline autonomy/cli/helpers/chain.py:64]: We don't need funded keys for every on chain interaction

**Comment by angrybayblade** [comment]
I'll update the chain selection process for `from-token` command in the docs PR

**Review by 0xArdi — APPROVED** [review]

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**